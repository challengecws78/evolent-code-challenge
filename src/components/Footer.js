import React from 'react';
import { NavLink } from 'react-router-dom';

export const Footer = () => (
  <footer>
    Â© 2018
  </footer>
);

export default Footer;