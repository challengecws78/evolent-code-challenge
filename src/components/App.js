import React, { Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import DynamicImport from './DynamicImport';
import Header from './Header';
import Footer from './Footer';

const ContactPage = (props) => (
  <DynamicImport load={() => import('./Contacts/ContactPage')}>
    {(Component) => Component === null
      ? <span></span>
      : <Component {...props} />}
  </DynamicImport>
)

const AddContact = (props) => (
  <DynamicImport load={() => import('./Contacts/AddContact')}>
    {(Component) => Component === null
      ? <span></span>
      : <Component {...props} />}
  </DynamicImport>
)

const EditContact = (props) => (
  <DynamicImport load={() => import('./Contacts/EditContact')}>
    {(Component) => Component === null
      ? <span></span>
      : <Component {...props} />}
  </DynamicImport>
)

const App = () => (
  <Router>
    <Fragment>
      <Header />
      <Switch>
        <Route path="/" component={ContactPage} exact={true} />
        <Route path="/create" component={AddContact} />
        <Route path="/edit/:id" component={EditContact} />
      </Switch>
      <Footer />
    </Fragment>
  </Router>
);

export default App;
