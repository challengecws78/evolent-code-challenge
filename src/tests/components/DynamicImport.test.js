import React from 'react';
import { shallow } from 'enzyme';
import { DynamicImport } from '../../components/DynamicImport';



test('Will render with a component if there is a component imported', () => {
    let wrapper = shallow(
        <DynamicImport load={() => import('../../../src/components/Header')}>
            {(Component) => Component === null
            ? <span></span>
            : <Component {...props} />}
        </DynamicImport>
    )
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('component')).toBe(null)
});