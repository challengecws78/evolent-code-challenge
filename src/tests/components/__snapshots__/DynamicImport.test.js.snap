// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Will render with a component if there is a component imported 1`] = `<span />`;
